coderd/database/queries/aibridge.sql

```diff
@@ -1,40 +1,160 @@
   id, interception_id, provider_response_id, prompt, metadata, created_at
 ) VALUES (
   @id, @interception_id, @provider_response_id, @prompt, COALESCE(@metadata::jsonb, '{}'::jsonb), @created_at
-);
+)
+RETURNING *;
 
--- name: InsertAIBridgeToolUsage :exec
+-- name: InsertAIBridgeToolUsage :one
 INSERT INTO aibridge_tool_usages (
   id, interception_id, provider_response_id, tool, server_url, input, injected, invocation_error, metadata, created_at
 ) VALUES (
   @id, @interception_id, @provider_response_id, @tool, @server_url, @input, @injected, @invocation_error, COALESCE(@metadata::jsonb, '{}'::jsonb), @created_at
-);
+)
+RETURNING *;
 
 -- name: GetAIBridgeInterceptionByID :one
-SELECT * FROM aibridge_interceptions WHERE id = @id::uuid;
+SELECT
+	*
+FROM
+	aibridge_interceptions
+WHERE
+	id = @id::uuid;
 
 -- name: GetAIBridgeInterceptions :many
-SELECT * FROM aibridge_interceptions;
+SELECT
+	*
+FROM
+	aibridge_interceptions;
 
 -- name: GetAIBridgeTokenUsagesByInterceptionID :many
-SELECT * FROM aibridge_token_usages WHERE interception_id = @interception_id::uuid;
+SELECT
+	*
+FROM
+	aibridge_token_usages WHERE interception_id = @interception_id::uuid
+ORDER BY
+	created_at DESC,
+	id DESC;
 
 -- name: GetAIBridgeUserPromptsByInterceptionID :many
-SELECT * FROM aibridge_user_prompts WHERE interception_id = @interception_id::uuid;
+SELECT
+	*
+FROM
+	aibridge_user_prompts
+WHERE
+	interception_id = @interception_id::uuid
+ORDER BY
+	created_at DESC,
+	id DESC;
 
 -- name: GetAIBridgeToolUsagesByInterceptionID :many
-SELECT * FROM aibridge_tool_usages WHERE interception_id = @interception_id::uuid;
+SELECT
+	*
+FROM
+	aibridge_tool_usages
+WHERE
+	interception_id = @interception_id::uuid
+ORDER BY
+	created_at DESC,
+	id DESC;
+
+-- name: ListAIBridgeInterceptions :many
+SELECT
+	*
+FROM
+	aibridge_interceptions
+WHERE
+	-- Filter by time frame
+	CASE
+		WHEN @started_after::timestamptz != '0001-01-01 00:00:00+00'::timestamptz THEN aibridge_interceptions.started_at >= @started_after::timestamptz
+		ELSE true
+	END
+	AND CASE
+		WHEN @started_before::timestamptz != '0001-01-01 00:00:00+00'::timestamptz THEN aibridge_interceptions.started_at <= @started_before::timestamptz
+		ELSE true
+	END
+	-- Filter initiator_id
+	AND CASE
+		WHEN @initiator_id::uuid != '00000000-0000-0000-0000-000000000000'::uuid THEN aibridge_interceptions.initiator_id = @initiator_id::uuid
+		ELSE true
+	END
+	-- Filter provider
+	AND CASE
+		WHEN @provider::text != '' THEN aibridge_interceptions.provider = @provider::text
+		ELSE true
+	END
+	-- Filter model
+	AND CASE
+		WHEN @model::text != '' THEN aibridge_interceptions.model = @model::text
+		ELSE true
+	END
+	-- Cursor pagination
+	AND CASE
+		WHEN @after_id::uuid != '00000000-0000-0000-0000-000000000000'::uuid THEN (
+			-- The pagination cursor is the last ID of the previous page.
+			-- The query is ordered by the started_at field, so select all
+			-- rows before the cursor and before the after_id UUID.
```

Comment:
Nit: we could add here that we're sorting DESC which is why we're searching _before_ the _after_ ID.
The naming of "after ID" makes this quite confusing; we should consider something else like `from_id` or `interception_cursor_id` or something.